### PR TITLE
fix(ci): Switched from macOS runners to Ubuntu runners

### PR DESCRIPTION
**Pull request by akhilamoyila9**
**changes:**
1) "Changed the macOS-12-large runner label to ubuntu-latest"
   **reason**: "Linux runners are more efficient for nested virtualization"
 2) "Updated python-version to 3.10.12"
    **reason**: "Compatible with the latest Ubuntu runner"
3) "Added steps for installing VirtualBox and Vagrant"
    **reason**: "Required for virtualization setup"
4) "Implemented maximize build space action"
    **reason**: "To optimize available resources and free up space"

## Additional Information

- [ ] This change is backwards-breaking

### CODE REVIEW

**Comment by maxhbr** [comment]
>  reason: "Linux runners are more efficient for nested virtualization"

can you elaborate?

**Comment by akhilamoyila9** [comment]
@maxhbr Due to issues with macOS 12 images producing VirtualBox machines in Gurumeditation state, Got the suggestion from runner images community to use linux runners as they are good at nested virtualization , attaching the comment link here https://github.com/actions/runner-images/issues/8730#issuecomment-1824248825

File: .github/workflows/lte-integ-test-bazel-magma-deb.yml [hunk 1]
```diff
@@ -51,11 +52,15 @@ jobs:
 
       - uses: actions/setup-python@13ae5bb136fac2878aff31522b9efb785519f984 # pin@v4.3.0
         with:
-          python-version: '3.8.10'
+          python-version: '3.10.12'
       - name: Install pre requisites
         run: |
           pip3 install --upgrade pip
           pip3 install ansible fabric jsonpickle requests PyYAML firebase_admin
+          sudo apt update && sudo apt install virtualbox
```
Comment (tmdzk) [inline .github/workflows/lte-integ-test-bazel-magma-deb.yml:60]: Wait nested virtualization is enabled on Ubuntu nodes?

Comment (akhilamoyila9) [reply .github/workflows/lte-integ-test-bazel-magma-deb.yml:60]: Yes, @tmdzk , nested virtualization is indeed enabled on the Ubuntu nodes. This enables us to install VirtualBox within our workflow, allowing us to utilize virtualization capabilities within our virtual environments.

Comment (tmdzk) [reply .github/workflows/lte-integ-test-bazel-magma-deb.yml:60]: That's a great idea, I'm all in!

**Review by tmdzk — APPROVED** [review]

**Pull request merged**